src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -195,6 +196,18 @@ var ReactCompositeComponentMixin = {
           Component.displayName || Component.name || 'Component'
         );
       }
+
+      var propsMutated = !shallowEqual(inst.props || {}, publicProps);
```

Comment:
By ===, you mean deep comparison of objects? I remember @jimfb asking me to avoid as it is expensive.